pkg/loki/modules.go

```diff
@@ -896,6 +896,12 @@ func (t *Loki) initIndexGateway() (services.Service, error) {
 }
 
 func (t *Loki) initIndexGatewayRing() (_ services.Service, err error) {
+	// IndexGateway runs by default on read target, and should always assume
```

Comment:
I didn't get this part: why it should always run in ring mode?

Reply:
if you don't run in ring mode, then there's no difference between running the index gateway or not. when there's no index gateway the queriers fetch and cache indexes locally, and since they're not clustered you end up duplicating the indexes on every querier. if you run the index gateway on every read mode without sharding, you end up in the same place with every index being duplicated across every read node. therefore, there doesn't seem to be any good reason to enable the index gateway on every read node unless we also run them in ring mode.

Reply:
That's a good point. What are the problems with the default ring configs, what configs could we check to make sure they're setup correctly?

Reply:
Ex:
* Since the default indexgateway mode is simple, let's imagine I don't configure anything for it
* But let's suppose I define a common instance_addr but no common ring
* And finally, I configure all my other rings to run etcd.

In that scenario, the indexgatewayring will actually run and try to use consul which isn't configured.